analyzers/src/SonarAnalyzer.CSharp/Rules/GenericReadonlyFieldPropertyAssignment.cs

```diff
@@ -62,45 +61,40 @@ protected override void Initialize(SonarAnalysisContext context)
                 SyntaxKind.RightShiftAssignmentExpression,
                 SyntaxKindEx.CoalesceAssignmentExpression);
 
-            context.RegisterSyntaxNodeActionInNonGenerated(
-                    c =>
-                    {
-                        var unary = (PrefixUnaryExpressionSyntax)c.Node;
-                        var expression = unary.Operand;
-
-                        ProcessPropertyChange(expression, c.SemanticModel, c);
-                    },
-                    SyntaxKind.PreDecrementExpression,
-                    SyntaxKind.PreIncrementExpression);
+            context.RegisterSyntaxNodeActionInNonGenerated(c =>
+                {
+                    var unary = (PrefixUnaryExpressionSyntax)c.Node;
+                    ProcessPropertyChange(c, c.SemanticModel, unary.Operand);
+                },
+                SyntaxKind.PreDecrementExpression,
+                SyntaxKind.PreIncrementExpression);
 
-            context.RegisterSyntaxNodeActionInNonGenerated(
-                c =>
+            context.RegisterSyntaxNodeActionInNonGenerated(c =>
                 {
                     var unary = (PostfixUnaryExpressionSyntax)c.Node;
-                    var expression = unary.Operand;
-
-                    ProcessPropertyChange(expression, c.SemanticModel, c);
+                    ProcessPropertyChange(c, c.SemanticModel, unary.Operand);
                 },
                 SyntaxKind.PostDecrementExpression,
                 SyntaxKind.PostIncrementExpression);
         }
 
-        private static void ProcessPropertyChange(ExpressionSyntax expression, SemanticModel semanticModel, SyntaxNodeAnalysisContext context)
+        private static void ProcessPropertyChange(SyntaxNodeAnalysisContext context, SemanticModel semanticModel, ExpressionSyntax expression)
         {
-            if (!(expression is MemberAccessExpressionSyntax memberAccess)
-                || !(semanticModel.GetSymbolInfo(expression).Symbol is IPropertySymbol propertySymbol))
+            if (TupleExpressionSyntaxWrapper.IsInstance(expression))
             {
-                return;
+                foreach (var tupleArgument in ((TupleExpressionSyntaxWrapper)expression).Arguments)
+                {
+                    ProcessPropertyChange(context, semanticModel, tupleArgument.Expression);
+                }
             }
-
-            var fieldSymbol = semanticModel.GetSymbolInfo(memberAccess.Expression).Symbol as IFieldSymbol;
-            if (!IsFieldReadonlyAndPossiblyValueType(fieldSymbol)
-                || IsInsideConstructorDeclaration(expression, fieldSymbol.ContainingType, semanticModel))
+            else if (expression is MemberAccessExpressionSyntax memberAccess
+                && semanticModel.GetSymbolInfo(expression).Symbol is IPropertySymbol propertySymbol
```

Comment:
If I am not msitaken, this check `semanticModel.GetSymbolInfo(expression).Symbol is IPropertySymbol propertySymbol` can be the last one in the row so it will not be executed in case the IFieldSymbol does not comply with the other checks.